### DISCUSSION TITLE
Add extra credit if transaction is greater than some amount

### DISCUSSION THREAD

**Discussion by jigarbhaliya** [bavix/laravel-wallet#602] [Q&A] [answered]
I want to add extra credit to user wallet if transaction type is deposit and greater than amount x. 
I have intercepted  `TransactionCreatedEventInterface` with help of [this](https://bavix.github.io/laravel-wallet/#/event-customize) document.

As I am initiating new transaction (to add extra credit) inside listener of first transaction event, listener is called multiple times (first time for main transaction and second time for credit transaction). This results in incorrect user balance.

How to handle this? Any help will be appreciated.

**Comment by rez1dent3** [comment, marked as answer]
If we are talking about the credit limit, then you can not replenish the wallet, but increase the credit limit of the wallet.

https://github.com/bavix/laravel-wallet/blob/4ea8dada0f47fa973dbe24502fdc1a4c524c34b3/docs/credit-limits.md

Then, increase the value inside meta.

---

If you want to do the accrual of funds through the deposit method, then you can add meta data. Conditionally add "internal" to meta.
```php
$wallet->deposit(..., ['internal' => true]);
```
And already inside the event, check this field and if it exists, then interrupt the additional accrual.

```php
if ($transaction['meta']['internal'] ?? false) { return; }
```

Something like that.

**Reply by jigarbhaliya** [reply]
Thanks for the reply. I think it will work.
